pkg/kubernetes/node/v1alpha1/kubernetes.go

```diff
@@ -72,12 +93,24 @@ func KubeClient(opts ...kubeclientBuildOption) *Kubeclient {
 // getClientOrCached returns either a new instance
 // of kubernetes client or its cached copy
 func (k *Kubeclient) getClientOrCached() (*kubernetes.Clientset, error) {
+	var c *kubernetes.Clientset
+	var err error
 	if k.clientset != nil {
 		return k.clientset, nil
 	}
-	c, err := k.getClientset()
-	if err != nil {
-		return nil, errors.Wrapf(err, "failed to get clientset")
+	// KubeConfigPath holds the first priority to get clientset
+	if k.kubeConfigPath != "" {
+		c, err = k.getClientsetForPath(k.kubeConfigPath)
+		if err != nil {
+			return nil, errors.Wrapf(err,
+				"failed to get clientset kubeconfigpath: %s",
+				k.kubeConfigPath)
+		}
+	} else {
+		c, err = k.getClientset()
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to get clientset")
+		}
 	}
 	k.clientset = c
```

Comment:
In this case error handling should be done outside the if else block

Reply:
Intention is to log if error occurred because of wrong path.